tool/tctl/common/acl_command.go

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// ACLCommand implements the `tctl acl` family of commands.
+type ACLCommand struct {
+	format string
+
+	list        *kingpin.CmdClause
+	show        *kingpin.CmdClause
+	usersAdd    *kingpin.CmdClause
+	usersRemove *kingpin.CmdClause
+	usersList   *kingpin.CmdClause
+
+	// Used for managing a particular access list.
+	accessListName string
+
+	// Used for managing membership to an access list.
+	userName string
+	expires  string
+	reason   string
+}
+
+// Initialize allows ACLCommand to plug itself into the CLI parser
+func (c *ACLCommand) Initialize(app *kingpin.Application, _ *servicecfg.Config) {
+	acl := app.Command("acl", "Manage access lists.").Alias("access-lists")
+
+	c.list = acl.Command("list", "List cluster access lists.").Alias("ls")
+	c.list.Flag("format", "Output format, 'yaml', 'json', or 'text'").Default(teleport.YAML).EnumVar(&c.format, teleport.YAML, teleport.JSON, teleport.Text)
+
+	c.show = acl.Command("show", "Show detailed information for an access list..")
+	c.show.Arg("access-list-name", "The access list name.").Required().StringVar(&c.accessListName)
+	c.show.Flag("format", "Output format, 'yaml', 'json', or 'text'").Default(teleport.YAML).EnumVar(&c.format, teleport.YAML, teleport.JSON, teleport.Text)
+
+	users := acl.Command("users", "Manage user membership to access lists.")
+
+	c.usersAdd = users.Command("add", "Add a user to an access list.")
+	c.usersAdd.Arg("access-list-name", "The access list name.").Required().StringVar(&c.accessListName)
+	c.usersAdd.Arg("user-name", "The user to add to the access list.").Required().StringVar(&c.userName)
+	c.usersAdd.Arg("expires", "When the user's access expires (must be in RFC3339).").Required().StringVar(&c.expires)
```

Comment:
Should we also add some sort of `--ttl` flag? I'm thinking it woudl be much easier to say "make this valid for 6 months" than to have to calculate 6 months from today and convert that to RFC3339.

Also, most people aren't going to know what RFC3339 is without looking it up. Good CLIs tell people how to use them without making them look things up. Can we give an example?

Reply:
+1 on `-ttl` instead

Reply:
It will be a little bit awkward, because we usually use Go's `ParseDuration` for `--ttl` flags, and that only goes up to hours at the highest unit (I think).

Maybe `--months=6` or something instead.

Reply:
I'd vote for ttl just for more flexibility. Users may want less granularity than months.